app/recordtransfer/models.py

```diff
@@ -930,13 +952,23 @@ class InProgressSubmission(models.Model):
     )
     step_data = models.BinaryField(default=b"")
     title = models.CharField(max_length=256, null=True)
+    upload_session = models.ForeignKey(UploadSession, null=True, on_delete=models.SET_NULL)
 
     def clean(self) -> None:
-        """Validate the current step value."""
+        """Validate the current step value. This gets called when the model instance is
+        modified through a form.
+        """
         try:
             TransferStep(self.current_step)
         except ValueError:
             raise ValidationError({"current_step": ["Invalid step value"]}) from None
 
+    def upload_session_expires_at(self) -> Optional[timezone.datetime]:
+        """Get the expiration time of the upload session associated with this submission."""
+        if self.upload_session:
+            return self.upload_session.expires_at
+        return None
+
     def __str__(self):
+        """Return a string representation of this object."""
         return f"Transfer of {self.last_updated} by {self.user}"
```

Comment:
Adding a docstring here has highlighted that this str method isn't very good. How about this (open to suggestions):

```suggestion
    def __str__(self):
        """Return a string representation of this object."""
        title = self.title or "None"
        session = self.upload_session.token if self.upload_session else "None"
        return f"In-Progress Submission by {self.user} (Title: {title} | Session: {session})"
```

Reply:
Addressed by: https://github.com/NationalCentreTruthReconciliation/Secure-Record-Transfer/pull/402/commits/764dd18a4ac8ae5c37554ba3dc147e12ef192528